### DISCUSSION TITLE
How to use default structs in my own spec?

### DISCUSSION THREAD

**Discussion by benkeil** [kube-rs/kube#1297] [Q&A] [answered]
I want to use `k8s_openapi::api::core::v1::EnvVar` in my own spec, but the compiler says 

> the trait `JsonSchema` is not implemented for `EnvVar`

Is it possible to reuse it, or do I need to reimplement it by myself?
Or in my special case, can I skip it somehow (I reimplement a 3rd party CRD and don't want to create a CRD)?

**Comment by clux** [comment]
Hm, are you using the `schemars` feature on `k8s-openapi`: https://github.com/Arnavion/k8s-openapi/blob/52da26009c948cf02e82376212aee2ff63158590/Cargo.toml#L39 ?

**Reply by benkeil** [reply, marked as answer]
figured it out:

```rust
#[derive(Clone, Serialize, Deserialize, Debug, JsonSchema)]
#[serde(rename_all = "camelCase")]
pub struct TaskOption {
    pub r#for: Vec<String>,
    #[schemars(skip_serializing, skip_deserializing)]
    pub env: Option<Vec<EnvVar>>,
    #[schemars(skip_serializing, skip_deserializing)]
    pub env_from: Option<Vec<EnvFromSource>>,
}
```
